src/plugins/data/public/ui/dataset_selector/configurator.tsx

```diff
@@ -53,17 +57,20 @@ export const Configurator = ({
 
   useEffect(() => {
     const fetchFields = async () => {
-      const datasetFields = await queryString
-        .getDatasetService()
-        .getType(baseDataset.type)
-        ?.fetchFields(baseDataset);
+      const datasetFields =
+        baseDataset.type !== 'S3'
```

Comment:
I see, both PPL and SQL support aggregations. Not sure what's the product requirement here do we need time filter for S3 datasource? If not I remember @sejli  mentioned that we'll remove this all together for S3 datasources. If required, we can remove this line when we add in cache support as the subsequent call would just take it from cache.

Reply:
I do have a PR that should remove the time field selector if the language is SQL If this is to fetch fields on creation of the dataset, I don't think my change is related. 

Currently, we don't need time filter for S3 datasource because SQL doesn't support time filters.

Reply:
Updated the Datasets and DataTypeConfig to use a meta tag `requireTimefilter` This is now used to see, if we need to fetch fields for time filter or not.